src/thor/bidirectional_astar.cc

```diff
@@ -143,6 +143,10 @@ bool BidirectionalAStar::ExpandForward(GraphReader& graphreader,
   if (!costing_->Allowed(nodeinfo)) {
     const DirectedEdge* opp_edge;
     const GraphId opp_edge_id = graphreader.GetOpposingEdgeId(pred.edgeid(), opp_edge, tile);
+    // Mark the predecessor as a deadend to be consistent with how the
+    // edgelabels are set when an *actual* deadend (i.e. some dangling OSM geometry)
+    // is labelled
+    pred.set_deadend(true);
```

Comment:
nice catch, we totally forgot to do that!

Reply:
Took a little digging to understand the usage of the deadend flag, but it was a one-liner (per algorithm!) in the end.

@kevinkreiser I also fixed timedep_forward/reverse - are there any other instances that likely need this?

When is `dijkstra.cc` used?  It doesn't have any of the work from https://github.com/valhalla/valhalla/pull/2705 - was that deliberate?

Reply:
i was going to mention those two but saw you already got them :+1: 

dijkstras is used in the reach check when the faster conservative reach check fails to find minimum reachability. the lack of derived deadend support in that algorithm is definitely an oversight.

i think its ok to note it in an issue though and come back to it in a separate PR if you would prefer it (tests will change because of it so it might be a decent sized rabbit hole). specifically we now have two types of deadend per search tree:

1. deadends because of node access which you just fixed or completed :wink: 
2. deadends because of edge access which was already there from a long time ago refactor @PureW and i did to fix other no route scenarios
 
both should make it into dijkstras to make the algorithm robust in the way the others are but like i said it might be a bit of a task to wade through all the test changes so maybe its better as a separate issue/PR?

Reply:
Opened a ticket to review all this stuff - I think we need some consolidation. https://github.com/valhalla/valhalla/issues/2885